preparedstatement/src/main/java/software/amazon/athena/preparedstatement/CreateHandler.java

```diff
@@ -0,0 +1,79 @@
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class CreateHandler extends BaseHandlerStd {
+    private Logger logger;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final CallbackContext callbackContext,
+        final ProxyClient<AthenaClient> proxyClient,
+        final Logger logger) {
+
+        this.logger = logger;
+
+        return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
+
+            .then(progress ->
+                proxy.initiate("AWS-Athena-PreparedStatement::Create::PreExistanceCheck",
+                    proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+
+                    .translateToServiceRequest(Translator::translateToReadRequest)
+
+                    .makeServiceCall((awsRequest, client) -> {
+                        client.injectCredentialsAndInvokeV2(
+                            awsRequest, client.client()::getPreparedStatement);
+                        logger.log(String.format("%s has been read successfully.", ResourceModel.TYPE_NAME));
+                        throw new CfnAlreadyExistsException(ResourceModel.TYPE_NAME, awsRequest.statementName());
+                    })
+
+                    .handleError((awsRequest, exception, client, model, context) -> {
+                        if (exception instanceof ResourceNotFoundException) { // no preparedstatement
+                            return ProgressEvent.progress(model, context);
+                        } else if (exception instanceof InvalidRequestException &&
+                            exception.getMessage().contains("is not found")) { // no workgroup
+                            // if primary workgroup is not found, it could be
+                            // created during preparedStatement creation
+                            return ProgressEvent.progress(model, context);
+                        }
+                        throw exception;
+                    })
+                    .progress()
+            )
+
+            .then(progress ->
+                proxy.initiate("AWS-Athena-PreparedStatement::Create",
+                    proxyClient,progress.getResourceModel(), progress.getCallbackContext())
+                    .translateToServiceRequest(Translator::translateToCreateRequest)
+                    .makeServiceCall((awsRequest, client) -> {
+                        CreatePreparedStatementResponse awsResponse;
+                        try {
+                          awsResponse = client.injectCredentialsAndInvokeV2(
+                              awsRequest, client.client()::createPreparedStatement);
+                        } catch (final AwsServiceException e) {
```

Comment:
Do we want to add log's for failure scenario's?

Reply:
we could add them, but the AwsServiceException that's thrown does contain error messages. Not sure what additional information we want to log.